src/app/modules/main/chat_section/module.nim

```diff
@@ -679,20 +679,15 @@ proc addNewChat(
       memberRole = community.memberRole
 
   var categoryOpened = true
-  if chatDto.categoryId != "":
-    let categoryItem = self.view.chatsModel.getItemById(chatDto.categoryId)
-    categoryOpened = categoryItem.categoryOpened
-    if channelGroup.id != "":
-      for category in channelGroup.categories:
-        if category.id == chatDto.categoryId:
-          categoryPosition = category.position
-          break
-    else:
-      let category = self.controller.getCommunityCategoryDetails(self.controller.getMySectionId(), chatDto.categoryId)
-      if category.id == "":
-        error "No category found for chat", chatName=chatDto.name, categoryId=chatDto.categoryId
-      else:
-        categoryPosition = category.position
+
+  let category = self.controller.getCommunityCategoryDetails(self.controller.getMySectionId(), chatDto.categoryId)
+  if category.id == "":
+    error "No category found for chat", chatName=chatDto.name, categoryId=chatDto.categoryId
+  else:
+    categoryOpened = category.categoryOpened
+    categoryPosition = category.position
+
+    self.view.chatsModel().changeCategoryOpened(category.id, categoryOpened)
```

Comment:
hum... Without this model update, I got unwanted artefacts on the category button state `>` or `v` after boot. This seems required no? Because it makes sure the model is aligned.

Reply:
Ok, yeah that would fix it, but there has to be a better way. Because this way updates the model for every channel in a category.

I think this call should be in `addCategoryItem` when we add a category.